### PR TITLE
Add centralized method to fetch Bundles for a particular Cluster

### LINKED ISSUE #313: Get currently installed `VersionsBundle`
In order to compare the latest available component versions (from latest `Bundles` manifest), we need to retrieve the `VersionsBundle` that corresponds to the current state of the cluster. This should be available in the cluster. The controller already does this.

Find a common to way to retrieve this bundle for both the cli and the controller.

### PR DESCRIPTION
**Pull request by g-gaston**
Resolves #313 

*Description of changes:*
Move logic to get `Bundles` and build a cluster spec for a certain `Cluster` to the `cluster` package

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/363#" title="Author self-approved">g-gaston</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/executables/kubectl_test.go
Comment (g-gaston) [inline pkg/executables/kubectl_test.go:665]: I just moved this to the top of the file

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/363#" title="Author self-approved">g-gaston</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/363#pullrequestreview-769404076" title="LGTM">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [g-gaston,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
